## 双十一价格的统计描述

因为我们只根据抓到的所有商品做统计描述，那么由于商品数量的原因，我们选的各类商品很难保持和整个平台的商品种类分布一致，因此考虑结果有一定误差。

### 双十一价格描述
+ 双十一价格平均下调8%
+ 最大折扣为90.5%
+ 43.3%的商品价格相对于10号没有变动
+ 55.0%的商品价格相对于10号下降
+ 1.7%的商品价格相对于10号上涨

下面为简单的统计描述：

In [7]:
price_change['11-11'].describe()

count    41133.000000
mean        -0.086311
std          0.180490
min         -0.905000
25%         -0.131034
50%         -0.075019
75%          0.000000
max         12.181818
Name: 11-11, dtype: float64

### 双十一之后十二号价格变动
+ 十二号价格相对于十号价格打折1%，说明十二号仍存在部分商品保持双十一的价格。
+ 侧面反映平台活动实际上，降低了活动日前后的价格
+ 87.7%的价格在十二号回调至十号的价格
+ 9.0%的价格在十二号仍然保持十一号大减价的状态

下面为简单的统计描述：


In [6]:
price_change['11-12'].describe()

count    41133.000000
mean        -0.010564
std          0.146130
min         -0.989000
25%          0.000000
50%          0.000000
75%          0.000000
max         12.181818
Name: 11-12, dtype: float64

### 食品
+ 食品在双十一整体折扣幅度为7.4%，比总体的8%略低。
+ 双十一最高折扣为88.8%。
+ 双十一食品中，50.8%的价格下降，46.5%的价格保持不变。
+ 十二号价格变动中，8.4%的商品继续保持双十一打折的状态，87.7%的商品回调至十号的价格。

下面为简单的统计描述：

In [18]:
cate1['11-11'].describe()

count    22467.000000
mean        -0.074641
std          0.181042
min         -0.887500
25%         -0.123596
50%         -0.010101
75%          0.000000
max          4.378151
Name: 11-11, dtype: float64

In [19]:
cate1['11-12'].describe()

count    22467.000000
mean        -0.004606
std          0.121025
min         -0.864583
25%          0.000000
50%          0.000000
75%          0.000000
max          3.962406
Name: 11-12, dtype: float64

### 服装
+ 服装在双十一整体折扣幅度为15.8%，基本是总体的8%的两倍，说明双十一打折的的主力商品是服装。
+ 双十一最高折扣为82.1%。
+ 双十一服装中，90.9%的价格下降，8.9%的价格保持不变。
+ 十二号价格变动中，14.1%的商品继续保持双十一打折的状态，83.6%的商品回调至十号的价格。

下面为简单的统计描述：

In [20]:
cate3['11-11'].describe()

count    2169.000000
mean       -0.158460
std         0.125231
min        -0.821429
25%        -0.190217
50%        -0.126316
75%        -0.102041
max         1.900000
Name: 11-11, dtype: float64

In [21]:
cate3['11-12'].describe()

count    2169.000000
mean       -0.027125
std         0.112610
min        -0.793697
25%         0.000000
50%         0.000000
75%         0.000000
max         1.000000
Name: 11-12, dtype: float64

## 医疗
+ 医疗在双十一整体折扣幅度为15.7%，基本是总体的8%的两倍，同样为价格大幅度下降的商品，
+ 双十一最高折扣为89.0%。
+ 双十一医疗中，82.5%的价格下降，17.0%的价格保持不变。
+ 十二号价格变动中，12.6%的商品继续保持双十一打折的状态，86.1%的商品回调至十号的价格。

下面为简单的统计描述：

In [22]:
cate5['11-11'].describe()

count    1585.000000
mean       -0.157080
std         0.138846
min        -0.869565
25%        -0.213759
50%        -0.129167
75%        -0.100143
max         1.253133
Name: 11-11, dtype: float64

In [23]:
cate5['11-12'].describe()

count    1585.000000
mean       -0.026228
std         0.100063
min        -0.855072
25%         0.000000
50%         0.000000
75%         0.000000
max         0.769231
Name: 11-12, dtype: float64

In [17]:
# -*- coding: utf-8 -*-
"""
Created on Thu Dec 08 11:16:16 2016

@author: Administrator
"""
import pandas as pd
import numpy as np

# read document as dataframe
table = pd.read_csv(r'C:\Users\Administrator\Desktop\double11\company')

# get cate of all goods
quatity_table = table.iloc[:,[2,5]]
quatity_table = quatity_table.groupby(['goodsID'], as_index=True).mean()

# get the price of all goods
price_table = table.iloc[:,[2,6,7]]
price10 = price_table[price_table['date']=='2016-11-10']
price10 = price10.drop(['date'],1)
price10.columns = ['goodsID', '11-10']
price10 = price10.groupby(['goodsID'], as_index=True).mean()

price11 = price_table[price_table['date']=='2016-11-11']
price11 = price11.drop(['date'],1)
price11.columns = ['goodsID', '11-11']
price11 = price11.groupby(['goodsID'], as_index=True).mean()

price12 = price_table[price_table['date']=='2016-11-12']
price12 = price12.drop(['date'],1)
price12.columns = ['goodsID', '11-12']
price12 = price12.groupby(['goodsID'], as_index=True).mean()

price_all = pd.concat([quatity_table, price10, price11, price12], axis=1, join='inner')
price_all['big'] = price_all['catid'].apply(lambda x: str(x)[0:4])
price_all['medium'] = price_all['catid'].apply(lambda x: str(x)[0:6])
price_all['small'] = price_all['catid'].apply(lambda x: str(x)[0:8])

# get change of price
price_change = price_all.copy()
price_change['11-11'] = (price_all['11-11']-price_all['11-10'])/price_all['11-10']
price_change['11-12'] = (price_all['11-12']-price_all['11-10'])/price_all['11-10']

# describe cate3
cate3 = price_change[price_change['big']=='1103']

# describe cate1
cate1 = price_change[price_change['big']=='1101']

# describe cate5
cate5 = price_change[price_change['big']=='1105']